### DISCUSSION TITLE
when the shuffling of the trials happens?

### DISCUSSION THREAD

**Discussion by dokato** [jspsych/jsPsych#907] [Q&A] [answered]
Hey All,

sorry it wasn't clear to me from the documentation when exactly the shuffling of the trials happens. I want to split my experiment into 2 block sessions. In those session everything is the same - I just want to give a participant some break. But I would like to have my trials in different order in these blocks. For example, for this code:

```js
var block = {
    timeline: [
        fixation,
        trial,
        iti
    ],
    timeline_variables: conditions,
    sample: {
        type: 'fixed-repetitions',
        size: 5,
    }
};
timeline.push(block);
timeline.push(break);
timeline.push(block);
```

Does this keep the trial in the same order after and before the break, or it will shuffle that during every push I make here? Or it happens when I created the variable `block`?

**Comment by jodeleeuw** [comment, marked as answer]
Randomization should happen separately for each block in that case. The `sample` method is invoked right before the timeline runs (I think! It has been a while since I wrote that code...).
